### PR TITLE
parameterize build with PYTHON, PREFIX, SYSCONFDIR, DATADIR, and DESTDIR

### PR DESCRIPTION
**Pull request by kelnos**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,14 +1,30 @@
 # only really known to work on ubuntu, if you're using anything else, hopefully
 # it should at least give you a clue how to install it by hand
 # TODO: parameterize this and the xml file (maybe scons?)
```
Comment (salty-horse) [inline Makefile]: Remove this TODO? :)

Comment (kelnos) [inline Makefile]: hah, good call. done.

**Comment by rohieb** [comment]
I was stumbling over this issue after thinking myself of implementing this functionality in order to build a Debian package :) @salty-horse, if you're still interested, I have merged @kelnos' code with the current master, resolved the merge conflicts and polished it a bit, and pushed it to the branch `parameterize-build` at http://github.com/rohieb/ibus-uniemoji.

**Comment by salty-horse** [comment]
Thanks everyone for helping. Sorry for the delay in adding this obvious feature that many are asking for.

A few things I'm worried about:

1. sed's `s:@FOO@:$FOO:` will break if `$FOO` has `:` anywhere in it. Granted, it's an unlikely scenario for directory names. Is there a best-practice to avoid it?

2. The `uninstall` target `rm -rf`s some directories, which are now controlled by the caller. Could be dangerous. For safety, I think they should change to individual `rm -f` for each file, followed by an rmdir.

**Comment by rohieb** [comment]
On 17.09.2016 22:27, Ori Avtalion wrote:
>  1.
> 
>     sed's |s:@FOO@:$FOO:| will break if |$FOO| has |:| anywhere in it.
>     Granted, it's an unlikely scenario for directory names. Is there a
>     best-practice to avoid it?

Hmmm. I have looked at autoconf, which does weird things [involving leaning toothpicks](https://twitter.com/daniel_bohrer/status/777256870633439233), which I didn't quite understand. I would ignore that for now, given that it probably only affects 0.01% of users, and it errors out to the safe side if there are superfluous `:`.

```
$ make PREFIX='foo: bar' uniemoji.xml
sed -e "s:@PYTHON@:/usr/bin/python3:g; s:@DATADIR@:foo: bar/share:g" uniemoji.xml.in > uniemoji.xml
sed: -e expression #1, char 49: unknown option to `s'
Makefile:14: recipe for target 'uniemoji.xml' failed
make: *** [uniemoji.xml] Error 1
```

>  2.
> 
>     The |uninstall| target |rm -rf|s some directories, which are now
>     controlled by the caller. Could be dangerous. For safety, I think
>     they should change to individual |rm -f| for each file, followed by
>     an rmdir.

Right. I have done it like this and pushed to my branch.

 - Roland

**Comment by salty-horse** [comment]
Thanks!

I rebased it onto master and cleaned things up a bit:
* added an rm for uniemoji.py
* Changed `$(DESTDIR)$(PREFIX)/share` to `$(DESTDIR)$(DATADIR)`
* documented the PYTHON var
* removed extra spaces and a slash

**Pull request closed without merging**